bin/check_smoker_plugin.py

```diff
@@ -15,58 +15,109 @@
-            # No result?! -> UNKNOWN
-            nagios.exit_unknown("plugin has no last result")
-        elif plugin['lastResult']['status'] == 'OK':
-            try:
-                nagios.exit_ok(plugin['lastResult']['messages']['ok'][0])
-            except (KeyError, IndexError, TypeError):
-                nagios.exit_ok("smoke test %s succeeded at %s" % ( plugin['name'], plugin['lastResult']['lastRun']) )
-        elif plugin['lastResult']['status'] == 'WARN':
-            try:
-                nagios.exit_warning(plugin['lastResult']['messages']['warn'][0])
-            except (KeyError, IndexError, TypeError):
-                nagios.exit_warning(ParseIssue(plugin))
-        elif plugin['lastResult']['status'] == 'ERROR':
-            try:
-                nagios.exit_critical(plugin['lastResult']['messages']['error'][0])
-            except (KeyError, IndexError, TypeError):
-                nagios.exit_critical(ParseIssue(plugin))
+            results[UNKNOWN].append({'name': plugin_name, 'message': "plugin has no last result"})
         else:
-            nagios.exit_unknown("unknown status %s at %s" % (plugin['lastResult']['status'], plugin['lastResult']['lastRun']))
+            last_status = plugin['lastResult']['status']
+
+            if last_status in [ERROR, WARN, OK]:
+                try:
+                    results[last_status].append({'name': plugin_name,
+                                                 'message': plugin['lastResult']['messages'][last_status.lower()][0]})
+                except (KeyError, IndexError, TypeError):
+                    if last_status == OK:
+                        results[last_status].append(
+                            {'name': plugin_name,
+                             'message': "smoke test %s succeeded at %s" %
+                                        (plugin['name'], plugin['lastResult']['lastRun'])})
+                    else:
+                        results[last_status].append({'name': plugin_name, 'message': parse_issue(plugin)})
+            else:
+                results[UNKNOWN].append({'name': plugin_name,
+                                         'message': "unknown status %s at %s" %
+                                                    (plugin['lastResult']['status'], plugin['lastResult']['lastRun'])})
+
+    for status, exit_method in status_methods_pairs:
+        if results[status]:
+            if len(plugins.values()[0]['plugins']) == 1:
+                # if only one plugin has been executed, do not print summary
+                exit_method(results[status][0]['message'])
+            else:
+                summary = ', '.join(["%s: %s" % (s, len(results[s])) for s, _ in status_methods_pairs if results[s]])
+                messages = ['\n'.join(["%s - %s - %s" % (s, item['name'], item['message']) for item in list])
+                            for s, list in results.iteritems() if list]
+                exit_method("%s\n%s" % (summary, '\n'.join(messages)))
```

Comment:
Yes. The status_methods_pairs array is sorted according to severity in descending order, i.e. the script will end with ERROR if any plugin ended with ERROR, the script will end with WARN if any plugin ended with WARN and finally, the script will succeed if all plugins succeed.